CIME/XML/machines.py

```diff
@@ -318,14 +318,24 @@ def set_machine(self, machine, schema=None):
             return machine
         elif self.get_version() == 3:
             machines_file = os.path.join(
-                self.machines_dir, machine, "config_machines.xml"
+                os.environ.get("HOME"), ".cime", machine, "config_machines.xml"
```

Comment:
@jedwards4b Can we switch this to use `Pathlib`?

Reply:
We haven't been using Pathlib as a rule, but sure - do you want to do that or do you want me to?

Reply:
Part of the refactor (https://github.com/ESMCI/cime/issues/4528, still need to update with the exact tasks) I have planned is to migrate to Pathlib, now that our minimum supported python is above 3.4.

I can make the changes.